src/app/components/signing-dialog.component.ts

```diff
@@ -234,18 +254,67 @@ export class SigningDialogComponent {
     return summary.join('\n');
   }
 
-  signWithExtension() {
-    this.sign.emit({ signed: true, key: 'extension' });
+  async signWithExtension() {
+    this.validationError = '';
+    
+    try {
+      // Check if extension is available
+      if (!window.nostr) {
+        this.validationError = 'Nostr extension not found. Please install a Nostr extension.';
+        return;
+      }
+
+      // Get public key from extension
+      const extensionPubkey = await window.nostr.getPublicKey();
+      
+      // Validate that the extension's pubkey matches the expected pubkey
+      if (extensionPubkey !== this.expectedPubkey) {
+        this.validationError = 'The account signed in to the extension does not match the profile being edited. Please switch to the correct account in your extension or use the correct private key.';
+        return;
+      }
+
+      this.sign.emit({ signed: true, key: 'extension' });
+    } catch (error) {
+      console.error('Error getting public key from extension:', error);
+      this.validationError = 'Failed to get public key from extension. Please try again.';
+    }
   }
 
-  signWithPrivateKey() {
+  async signWithPrivateKey() {
+    this.validationError = '';
+    
     if (!this.privateKey) return;
-    this.sign.emit({ signed: true, key: this.privateKey });
-    this.privateKey = '';
+    
+    try {
+      let privateKeyHex = this.privateKey;
+      
+      // Convert nsec to hex if needed
+      if (this.privateKey.startsWith('nsec')) {
+        const decoded = nip19.decode(this.privateKey);
+        privateKeyHex = decoded.data as string;
+      }
+      
+      // Get public key from private key
+      const privateKeyBytes = new Uint8Array(privateKeyHex.match(/.{1,2}/g)!.map(byte => parseInt(byte, 16)));
```

Comment:
What is this use of regex on the private key hex? Should use an utility method instead that is either on noble libraries or nostr-tools, to convert hexToBytes, don't do that manually, because there are multiple pit-falls.